### PR TITLE
Constants tracker

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
This will make recurring constants review a lot easier since the diff of every PR to the constants will be very explicit.

---
This change is [Reviewable](https://reviewable.io/reviews/pathfinder-for-autonomous-navigation/flightsoftware/256)

### CODE REVIEW

File: src/common/constant_tracker.hpp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#ifndef CONSTANT_TRACKER_HPP_
+#define CONSTANT_TRACKER_HPP_
+
+#define TRACKED_CONSTANT(type, name, val) type name = val;
```
Comment (kkrol27) [inline src/common/constant_tracker.hpp]: To keep things less verbose we could potentially do:

    #define TRACKED_CONSTANT(type, name, val) static constexpr type name = val

I don't think anything would ever not be `static constexpr`.

File: tools/constant_reporter.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                        "val" : const[2].strip(),
+                        "file" : path
+                    })
+
+log = "Keep this file in version control so that changes to constants are well-known.\n\n"
+for const in consts:
+    log += f"{const['file']}: \"{const['name']}\" = \"{const['val']}\"\n"
+
+with open("constants", 'w') as f:
+    f.write(log)
```
Comment (kkrol27) [inline tools/constant_reporter.py]: Also, not at all necessary but I think we could do this in a 5 line bash script lol

**Comment by kkrol27** [comment]
One additional thought I had, we should design this with variables other than scalars in mind. For example, we'd want to be able to define gnc related constants like the prop system thruster matrix and such. Most likely we keep that as a `static constexpr lin::Matrix<float, 4, 3>` or something.

File: tools/constant_reporter.py (hunk 2 above)
Comment (tanishqaggarwal) [reply tools/constant_reporter.py]: This allows flexibility if we want to create CSVs or other formats for the constants.

File: src/common/constant_tracker.hpp (hunk 1 above)
Comment (tanishqaggarwal) [reply src/common/constant_tracker.hpp]: See `constants.hpp`; some constants can't be static. I could totally imagine wanting to create a non-constexpr constant as well (such as a vector of constants) so I feel like this is the minimum macro we can use.

I could add some more macros like TRACKED_CONSTANT_S, TRACKED_CONSTANT_SC, etc. to fix this

**Pull request merged**